mesonbuild/compilers/mixins/clike.py

```diff
@@ -1118,7 +1118,7 @@ def _find_library_real(self, libname: str, env: 'Environment', extra_dirs: T.Lis
         # Only try to find std libs if no extra dirs specified.
         # The built-in search procedure will always favour .so and then always
         # search for .a. This is only allowed if libtype is LibType.PREFER_SHARED
-        if ((not extra_dirs and libtype is LibType.PREFER_SHARED) or
+        if ((not extra_dirs and libtype in {LibType.PREFER_SHARED, LibType.SHARED}) or
```

Comment:
The commit message for this could do with being more descriptive about what the effect is and why we need to change the behavior.